eclair-core/src/test/scala/fr/acinq/eclair/payment/relay/NodeRelayerSpec.scala

```diff
@@ -323,6 +327,131 @@ class NodeRelayerSpec extends ScalaTestWithActorTestKit(ConfigFactory.load("appl
     register.expectNoMessage(100 millis)
   }
 
+  test("fail async payment when relay is not triggered") { f =>
+    val g = f.copy(nodeParams = f.nodeParams.copy(features = f.nodeParams.features.add(AsyncPaymentPrototype, Optional), relayParams = f.nodeParams.relayParams.copy(timeout = 5 seconds)))
+    import g._
+
+    val (nodeRelayer, _) = g.createNodeRelay(incomingAsyncPayment.head)
+    incomingAsyncPayment.foreach(p => nodeRelayer ! NodeRelay.Relay(p))
+
+    incomingAsyncPayment.foreach { p =>
+      val fwd = register.expectMessageType[Register.Forward[CMD_FAIL_HTLC]]
+      assert(fwd.channelId == p.add.channelId)
+      assert(fwd.message == CMD_FAIL_HTLC(p.add.id, Right(PaymentTimeout), commit = true))
+    }
+
+    register.expectNoMessage(100 millis)
+  }
+
+  test("fail async payment when only triggered before last incoming part received") { f =>
```

Comment:
You could use a test tag for all the async payment tests:
```suggestion
  test("fail async payment when only triggered before last incoming part received", Tag("async_payment")) { f =>
```

And when that test tag is present, in `withFixture` you would set the feature bit, so that you don't have to do it in every test.

Reply:
Great tip! I updated the tests in [fcdc394](https://github.com/ACINQ/eclair/pull/2435/commits/fcdc394cb2751d7408318afb437cf49d18389d08).